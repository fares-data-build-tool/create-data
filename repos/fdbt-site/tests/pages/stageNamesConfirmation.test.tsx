import * as React from 'react';
import { shallow } from 'enzyme';
import StageNamesConfirmation, {
    buildFareStageNamesConfirmationElements,
} from '../../src/pages/stageNamesConfirmation';

describe('pages', () => {
    describe('stageNamesConfirmation', () => {
        it('should render correctly', () => {
            const tree = shallow(
                <StageNamesConfirmation fareStageNames={['leeds', 'manchester', 'bolton']} csrfToken="" />,
            );
            expect(tree).toMatchSnapshot();
        });
    });
    describe('buildFareStageNamesConfirmationElements', () => {
        it('builds confirmation elements for an array of fare stages', () => {
            const result = buildFareStageNamesConfirmationElements(['leeds', 'manchester', 'bolton']);
            expect(result).toStrictEqual([
                { content: 'leeds', href: 'stageNames', name: 'Fare stage 1' },
                { content: 'manchester', href: 'stageNames', name: 'Fare stage 2' },
                { content: 'bolton', href: 'stageNames', name: 'Fare stage 3' },
            ]);
        });
    });
});
