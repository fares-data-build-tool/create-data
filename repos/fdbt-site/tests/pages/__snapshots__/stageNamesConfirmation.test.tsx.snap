// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages stageNamesConfirmation should render correctly 1`] = `
<Component
  description="Stage Names Confirmation page of the Create Fares Data Service"
  errors={Array []}
  title="Stage Names Confirmation - Create Fares Data Service"
>
  <CsrfForm
    action="/api/stageNamesConfirmation"
    csrfToken=""
    method="post"
  >
    <h1
      className="govuk-heading-l"
    >
      Check your Fare Stage Names before moving on
    </h1>
    <ConfirmationTable
      confirmationElements={
        Array [
          Object {
            "content": "leeds",
            "href": "stageNames",
            "name": "Fare stage 1",
          },
          Object {
            "content": "manchester",
            "href": "stageNames",
            "name": "Fare stage 2",
          },
          Object {
            "content": "bolton",
            "href": "stageNames",
            "name": "Fare stage 3",
          },
        ]
      }
      header="Fare Stage Names"
    />
    <input
      className="govuk-button"
      id="continue-button"
      type="submit"
      value="Continue"
    />
  </CsrfForm>
</Component>
`;
